prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -222,15 +220,29 @@ class FHIRRouter(
                 )
             )
 
-            // create translate event
-            val translateEvent = ProcessEvent(
-                Event.EventAction.TRANSLATE,
+            // create none event
+            var translateEvent = ProcessEvent(
```

Comment:
So, to better read this code, how about your create and if/else. One branch will use translate event. The other will use something like a noneEvent. Right now the reuse of the translateEvent variable confuses the flow a bit.  This will also clear up the multiple ifs you have even if you have to have a few lines of the same code.

Reply:
Code reworked as suggested in latest commit.